hca/staging_area_validator.py

```diff
@@ -0,0 +1,344 @@
+import argparse
+import base64
+from functools import (
+    cached_property,
+    lru_cache,
+)
+import json
+import logging
+from typing import (
+    List,
+    MutableMapping,
+    MutableSequence,
+    Optional,
+    Tuple,
+    TypeVar,
+)
+from urllib import (
+    parse,
+)
+import uuid
+
+import google.cloud.storage as gcs
+from jsonschema import (
+    FormatChecker,
+    ValidationError,
+    validate,
+)
+from more_itertools import (
+    one,
+)
+import requests
+
+T = TypeVar('T')
+JSON = MutableMapping[str, T]
+
+log = logging.getLogger(__name__)
+
+
+class StagingAreaValidator:
```

Comment:
This is the extraction of the `StagingAreaValidator` class from it's original home in `validate_staging_area.py`. It's almost a 100% copy except for the modification of the initializer to consume parameters rather than a list of args, and the removal of the `self.args.xyz` in favor of `self.xyz` instance variables.